assets/js/components/KeyMetrics/ChangeMetricsLink.js

```diff
@@ -48,7 +50,46 @@ export default function ChangeMetricsLink() {
 		trackEvent( `${ viewContext }_kmw`, 'change_metrics' );
 	}, [ setValue, viewContext ] );
 
-	if ( ! Array.isArray( keyMetrics ) || ! keyMetrics?.length > 0 ) {
+	const renderChangeMetricLink =
+		Array.isArray( keyMetrics ) || keyMetrics?.length > 0;
+
+	const isKeyMetricsSetupCompleted = useSelect( ( select ) =>
+		select( CORE_SITE ).isKeyMetricsSetupCompleted()
+	);
+
+	const keyMetricsSetupCompletedByUserID = useSelect( ( select ) =>
+		select( CORE_SITE ).getKeyMetricsSetupCompletedByUserID()
+	);
+
+	const currentUserID = useSelect( ( select ) =>
+		select( CORE_USER ).getID()
+	);
+
+	const { triggerOnDemandTour } = useDispatch( CORE_USER );
+
+	useEffect( () => {
+		const isUserEligibleForTour =
+			Number.isInteger( keyMetricsSetupCompletedByUserID ) &&
+			Number.isInteger( currentUserID ) &&
+			keyMetricsSetupCompletedByUserID > 0 &&
+			currentUserID !== keyMetricsSetupCompletedByUserID;
+
+		if (
+			renderChangeMetricLink &&
+			isKeyMetricsSetupCompleted &&
+			isUserEligibleForTour
+		) {
+			triggerOnDemandTour( sharedKeyMetrics );
```

Comment:
@techanvil @tofumatt Why did we decide to go with an on-demand tour? It seems like a good candidate for a normal "automatic" tour. All of this state for conditionally invoking the tour would be part of the tour's `checkRequirements` function, and it would automatically be triggered when the element is present. 🤔 

An on-demand trigger should really be triggered in response to an explicit action, rather than as an effect. Let me know if I'm missing something relevant here.

Reply:
@aaemnnosttv It was suggested by @techanvil here: https://github.com/google/site-kit-wp/issues/7346#issuecomment-1737009846. The code/logic to handle checking otherwise was actually a fair bit more complicated (see the issue history for my previous IBs without it), so we went with this option because it was much simpler to understand/implement.

Reply:
That makes sense, it would require additional logic to account for KMW showing if we were to remove it from here. In other tours such as the All Traffic widget, we'd used a flag in UI state to indicate when the widget was loaded as a similar case where we would otherwise need to leak a lot of detail into the tour's conditional check. If we went this route, the only change needed in this component would be to set that flag as a way around needing to duplicate conditions. The downside to this approach however is that it would be blocking in resolving the next tour and may never resolve if KMW isn't going to be shown, so it does seem better in this case to keep it as-is 👍